### PR TITLE
1.use 0.5.1 compiler, 2.bug fixing, and 3.align spec2.0

### PR DESCRIPTION
**Pull request by KimiWu123**
#1008

---

# Description
1. Using Solidity 0.5.1 to compile vrc
2. bug fixing, according to spec 2.0 Vyper sample code, `emit HashChainValue(receiptTree[1], depositParams, totalDepositCount);` should be `emit HashChainValue(receiptTree[1], depositData, totalDepositCount);`
3. Follow byte number in spec 2.0 (`msgGweiInBytes` and `timeStampInBytes` should be 8 bytes)
4. Using standard library function `abi.encodePacked` instead of `mergeBytes` to merge bytes, the gas consumption is almost the same.
[Spec 2.0 Vyper code](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#vyper-code)

### CODE REVIEW

File: contracts/validator-registration-contract/validator_registration.go [hunk 1]
```diff
@@ -5,21 +5,21 @@ package vrc
...
 	"math/big"
+	"prysm/prysm/shared/event"
 	"strings"
 
 	ethereum "github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/prysmaticlabs/prysm/shared/event"
```
Comment (prestonvanloon) [inline contracts/validator-registration-contract/validator_registration.go]: Revert this change?

Comment (KimiWu123) [reply contracts/validator-registration-contract/validator_registration.go]: Sorry, my miss. I have updated.

File: contracts/validator-registration-contract/validator_registration.sol [hunk 2]
```diff
@@ -16,14 +15,15 @@ contract ValidatorRegistration {
     );
 
     uint public constant DEPOSIT_SIZE = 32 ether;
-    uint public constant DEPOSITS_FOR_CHAIN_START = 8;
+    uint public constant DEPOSITS_FOR_CHAIN_START = 8; // 2**14
```
Comment (prestonvanloon) [inline contracts/validator-registration-contract/validator_registration.sol]: What does this comment mean? 2**14 is `16384`

Comment (KimiWu123) [reply contracts/validator-registration-contract/validator_registration.sol]: changed

**Comment by cyberbono3** [comment]
Following #1008, i was wrong stating that solidity produces float integers by division of two uint's. Instead, it results in floor of calculation with discarded remainder e.g. 17/2 = 8 (1 is discarded) 
Additionally, I suggest using Safemath library to prevent uint underflow/overflow  here:
 uint timestampDayBoundry = block.timestamp - block.timestamp % SECONDS_PER_DAY + SECONDS_PER_DAY;

**Comment by terenc3t** [comment]
@cyberbono3 That sounds good.  Can you please create an issue? and feel free to assign to yourself if you plan to work on it :)

**Comment by cyberbono3** [comment]
Sure)

**Comment by cyberbono3** [comment]
#1141
